### PR TITLE
Report on AMIs that are more than 90 days

### PR DESCRIPTION
**Pull request by cob16**
- Report on AMIs that are more than 90 days
- Second commit adds logging to audit events (required large refactor due to abstract class)

### CODE REVIEW

File: src/compliance/actionable_report_compliance.py [hunk 1]
```diff
@@ -17,7 +18,7 @@ def __init__(self, item_type: str, item: str):
         self.item = item
         self.action = BriefActionDescriber()
 
-    def analyse(self, audit: Audit) -> Set[Findings]:
+    def analyse(self, logger: Logger, audit: Audit) -> Set[Findings]:
```
Comment (aminebz949904) [inline src/compliance/actionable_report_compliance.py:21]: logger not used in the function

File: src/compliance/ec2_compliance.py [hunk 2]
```diff
@@ -1,10 +1,41 @@
...
-        return set()
+    def analyse(self, logger: Logger, audit: Audit) -> Set[Findings]:
+        findings = set()
+        for account in audit.report:
+            for instance in account["results"]["ec2_instances"]:
+                try:
+                    image_creation_date = datetime.fromisoformat(instance["image_creation_date"])
+                except ValueError:
+                    logger.error(
+                        f"'{instance['image_creation_date']}'does not match the date pattern"
```
Comment (aminebz949904) [inline src/compliance/ec2_compliance.py:22]: Adding instance id to the log may help

Comment (cob16) [reply src/compliance/ec2_compliance.py]: added this

**Review by aminebz949904 — APPROVED** [review]
looks good

**Comment by aminebz949904** [comment]
Looks good, I have some comments related to AuditAnalyser but not to do with your changes.

**Comment by achatain** [comment]
Why go through all these refactoring efforts when [getLogger()](https://docs.python.org/3/library/logging.html#logging.getLogger) essentially returns a singleton? I'd suggest to just call `getLogger()` where it's needed rather than pass it around 🤔 But what am I even doing here? 😅

**Comment by cob16** [comment]
@achatain your right that it may not necessary but I would argue its good to keep for 3 reasons:

1. If gives some felxabity to change to an alternate logger in the future. i.e in the future `configure_logging()` could output a non standard loggger such as for [JSON format logging](https://pypi.org/project/python-json-logger/) (this makes it easier to search in logging tools that can parse JSON)
2. It encourages developers to log if there is already one available 
3. We are trying embrace the principle of dependency injection here

**Pull request merged**